experimental/single_number_formatter/src/provider.rs

```diff
@@ -0,0 +1,92 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use alloc::borrow::Cow;
+use icu_provider::{yoke, zerofrom};
+use tinystr::UnvalidatedTinyAsciiStr;
+use zerovec::{VarZeroVec, ZeroMap};
+
+/// This type contains all of the essential data for currency formatting.
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+#[icu_provider::data_struct(CurrencyEssentialsV1Marker = "currency/essentials@1")]
+#[derive(Default, Clone, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_singlenumberformatter::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct CurrencyEssentialsV1<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub currency_patterns_map: ZeroMap<'data, UnvalidatedTinyAsciiStr<3>, CurrencyPatterns>,
+
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub standard: Cow<'data, str>,
+
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub standard_alpha_next_to_number: Cow<'data, str>,
+
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub place_holders: VarZeroVec<'data, str>,
+}
+
+#[zerovec::make_ule(PatternSelectionULE)]
+#[derive(Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq)]
+#[repr(u8)]
+pub enum PatternSelection {
+    /// Use the standard pattern.
+    Standard = 0,
+
+    /// Use the standard_alpha_next_to_number pattern.
+    StandardAlphaNextToNumber = 1,
+}
+
+type PatternSelectionMask = u8;
+
+// TODO(younies): replace this with Option<PlaceHolder>, enum PlaceHolder { Index(usize), ISO }
```

Comment:
Nit: I'm not going to block the PR on this, but in the future you should file issues for follow-up items, not just drop usernames in the code.